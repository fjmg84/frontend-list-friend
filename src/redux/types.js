export const TYPES = {
  LOAD: "load/friends",
  ERROR_ACTIVE: "error/active",
};
