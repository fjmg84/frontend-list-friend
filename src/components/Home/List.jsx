import React from 'react'
import PropTypes from 'prop-types'
import { useNavigate } from 'react-router-dom'
import { avatar } from '../../images/index'

export const List = ({ friends }) => {
	const navigate = useNavigate()

	return (
		<ul>
			{friends.map((friend) => {
				return (
					<li key={friend.id}>
						<div className="card">
							<div className="card-img">
								<img src={avatar} alt={friend.img} className="small" />
								<span
									className={`availability eclipse ${
										friend.available ? 'available' : 'not-available'
									}`}
								></span>
							</div>

							<div className="card-header">
								<div className="card-header-name">
									<p className="details-name">{`${friend.first_name} ${friend.last_name}`}</p>
								</div>
								<div className="card-header-status">
									<span className="details-status">{friend.status}</span>
								</div>
							</div>

							<div className="card-actions">
								<button
									className="details-button"
									onClick={() => navigate(`./show/${friend.id}`)}
									disabled={!friend.available}
								>
									Details
								</button>
							</div>
						</div>
					</li>
				)
			})}
		</ul>
	)
}

List.protoTypes = {
	friends: PropTypes.array,
}

List.propTypes = {
	friends: PropTypes.array.isRequired,
}
