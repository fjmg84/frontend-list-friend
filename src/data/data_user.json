{
    "id": 6,
    "img": "https://s3.amazonaws.com/uifaces/faces/twitter/walterstephanie/128.jpg",
    "first_name": "Steph",
    "last_name": "Walters",
    "phone": "(820) 289-1818",
    "address_1": "5190 Center Court Drive",
    "city": "Spring",
    "state": "TX",
    "zipcode": "77370",
    "bio": "I'm very choosy. I'm also very suspicious, very irrational and I have a very short temper. I'm also extremely jealous and slow to forgive. Just so you know.",
    "photos": [
        "/static/media/photo_1.65b44377ca3ba76ac460.jpg",
        "/static/media/photo_2.f94104513f3ac56066cc.jpg",
        "/static/media/photo_3.a617d38fcaf25508172f.jpg"
    ],
    "statuses": [
        "Developing something amazing",
        "This could be interesting....",
        "Man, life is so good",
        "There is nothing quite like a good friend",
        "Take a look around you, everything is awesome",
        "What is the point of all of this"
    ],
    "available": true
}