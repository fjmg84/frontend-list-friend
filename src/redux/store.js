import { createStore, combineReducers } from "redux";
import { friendsReducer } from "../reducer/friends";
import { errorsReducer } from "../reducer/errors";

const rootReducers = combineReducers({
  friendsReducer,
  errorsReducer,
});

export const store = createStore(
  rootReducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
