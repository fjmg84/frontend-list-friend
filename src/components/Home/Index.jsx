import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getFriends } from '../../redux/actions/friends'
import { setAlert } from '../../redux/actions/alerts'
import { TYPES } from '../../redux/types'
import { List } from './List'
import { Spinner } from '../Helps/Spinner'

export const Index = () => {
	const dispatch = useDispatch()
	const { status, msg, classAlert } = useSelector(
		(state) => state.alertsReducer,
	)
	const friends = useSelector((state) => state.friendsReducer)

	useEffect(() => {
		dispatch(setAlert(true, 'Loading friends...', 'loading'))
		getFriends()
			.then((friends) => {
				dispatch({ type: TYPES.LOAD, payload: friends })
				dispatch(setAlert(false, undefined))
			})
			.catch((err) => {
				console.log(err)
				dispatch(
					setAlert(true, 'Up!!! Sorry Not Found List Of Friends :(', 'error'),
				)
			})
	}, [dispatch])

	return (
		<div className="box">
			<div className="body">
				<div className="header">
					<h1>Friends</h1>
					{status && <div className={classAlert}>{msg}</div>}
				</div>

				<div className="main">
					{friends.length > 0 ? <List friends={friends} /> : <Spinner />}
				</div>
			</div>
		</div>
	)
}
