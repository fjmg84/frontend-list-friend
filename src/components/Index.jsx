import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { getFriends } from "../redux/actions/friends";
import { setAlert } from "../redux/actions/alerts";
import { TYPES } from "../redux/types";
import { List } from "./List";

export const Index = () => {
  const dispatch = useDispatch();
  const { status, msg, classAlert } = useSelector(
    (state) => state.alertsReducer
  );

  useEffect(() => {
    dispatch(setAlert(true, "Loading friends...", "loading"));
    getFriends()
      .then((friends) => {
        dispatch({ type: TYPES.LOAD, payload: friends });
        dispatch(setAlert(false, undefined));
      })
      .catch((err) => {
        console.log(err);
        dispatch(setAlert(true, "Up!! Sorry. Not loading Friends :(", "error"));
      });
  }, [dispatch]);

  return (
    <div className="box">
      <div className="body">
        <div className="header">
          <h1>Friends</h1>
          {status && <div className={classAlert}>{msg}</div>}
        </div>

        <div className="main">
          <List />
        </div>
      </div>
    </div>
  );
};
