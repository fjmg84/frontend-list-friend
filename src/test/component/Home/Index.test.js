import React from "react";
import { MemoryRouter } from "react-router-dom";
import { render } from "enzyme";
import { Provider } from "react-redux";
import { Index } from "../../../components/Home/Index";
import { store } from "../../../redux/store";
import data from "../../../data/data.json";
import { TYPES } from "../../../redux/types";

describe("testing component <Index/>", () => {
  test("snapshot to component <Index/>", () => {
    store.dispatch({
      type: TYPES.LOAD,
      payload: data,
    });

    const wrapper = render(
      <MemoryRouter>
        <Provider store={store}>
          <Index />
        </Provider>
      </MemoryRouter>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
