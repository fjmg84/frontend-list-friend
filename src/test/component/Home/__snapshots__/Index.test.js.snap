// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing component <Index/> snapshot to component <Index/> 1`] = `
<div
  class="box"
>
  <div
    class="body"
  >
    <div
      class="header"
    >
      <h1>
        Friends
      </h1>
    </div>
    <div
      class="main"
    >
      <ul>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/vladarbatov/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse not-available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Jeremy Davis
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  At work...
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
                disabled=""
              >
                Details
              </button>
            </div>
          </div>
        </li>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/vladarbatov/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Vlad Baratovich
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  Hangout out by the pool
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
              >
                Details
              </button>
            </div>
          </div>
        </li>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/rssems/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Reese Samsonite
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  At NG-conf!
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
              >
                Details
              </button>
            </div>
          </div>
        </li>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/enda/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse not-available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Edwardo Gonzalez
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  Developing something amazing
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
                disabled=""
              >
                Details
              </button>
            </div>
          </div>
        </li>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/jm_denis/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse not-available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Jim Denison
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  Designing beatiful things
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
                disabled=""
              >
                Details
              </button>
            </div>
          </div>
        </li>
        <li>
          <div
            class="card"
          >
            <div
              class="card-img"
            >
              <img
                alt="https://s3.amazonaws.com/uifaces/faces/twitter/walterstephanie/128.jpg"
                class="small"
                src="photo.jpg"
              />
              <span
                class="availability eclipse available"
              />
            </div>
            <div
              class="card-header"
            >
              <div
                class="card-header-name"
              >
                <p
                  class="details-name"
                >
                  Steph Walters
                </p>
              </div>
              <div
                class="card-header-status"
              >
                <span
                  class="details-status"
                >
                  Developing something amazing
                </span>
              </div>
            </div>
            <div
              class="card-actions"
            >
              <button
                class="details-button"
              >
                Details
              </button>
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
